### PR TITLE
Added Pre-Build process

### PR DESCRIPTION
**Pull request by vbiamitr**

### CODE REVIEW

File: bin/pbiviz-package.js [hunk 1]
```diff
@@ -64,11 +63,15 @@ if (!program.pbiviz && !program.resources) {
...
-    PreProcessor.processPbiviz(visualPackage, program);    
-
     let buildOptions = {
         minify: program.minify,
-        plugin: program.plugin || program.pbiviz
+        plugin: program.plugin || program.pbiviz,
+        preProcessOptions: {
+            tag: program.tag,
+            withGuidPrefix: program.withGuidPrefix,
+            buildVersion: program.buildVersion,
```
Comment (shoaibmerchant) [inline bin/pbiviz-package.js]: we removed build version, i think you missed it here

File: lib/PreProcessor.js [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+    * @param {Object} buildOptions
+    */
+   static processPbiviz(visualPackage, buildOptions) {
+        // guid
+        if(buildOptions.withGuidPrefix) {
+            visualPackage.config.visual.guid = `${buildOptions.withGuidPrefix}_${visualPackage.config.visual.guid}`;
+        }
+
+        // visual version
+        if(buildOptions.buildVersion) {
```
Comment (shoaibmerchant) [inline lib/PreProcessor.js]: again I thought we were removing this?

Comment (shoaibmerchant) [reply lib/PreProcessor.js]: and instead setting the placeholder (buildVersion) using the visualPackage.config.visual.version

Comment (vbiamitr) [reply lib/PreProcessor.js]: PR updated

File: bin/pbiviz-package.js (hunk 1 above)
Comment (vbiamitr) [reply bin/pbiviz-package.js]: yeah.. removed it

**Pull request merged**